### PR TITLE
Fixed :class:`~.Rotate` for angles that are multiples of :math:`2\pi`

### LINKED ISSUE #1935: `Rotate` breaks when rotating multiples of 2pi
## Description of bug / unexpected behavior
```py
class RotationAnim(Scene):
    def construct(self):
        a = Circle().move_to(RIGHT)
        self.play(Rotate(a, 2*PI, about_point=ORIGIN), run_time=5)
```

gives:

https://user-images.githubusercontent.com/9102529/130183273-e101d2fe-65cc-42af-80ee-82d3adb0c732.mp4

`Rotating` works, but does not take into account the `rate_func` (and according to @behackl, is probably intentional)? Perhaps the best way to fix this is to use the method used for `Rotating`, just taking into account the `rate_func` (and maybe `Rotating` can inherit from `Rotate` and simply ignore the rate_func).

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
Paste your code here.
```

</details>

## Additional media files

<details><summary>Images/GIFs</summary>

</details>

## Logs
<details><summary>Terminal output</summary>

```
PASTE HERE OR PROVIDE LINK TO https://pastebin.com/ OR SIMILAR
```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (with version, e.g Windows 10 v2004 or macOS 10.15 (Catalina)):
- RAM:
- Python version (`python/py/python3 --version`):
- Installed modules (provide output from `pip list`):
```
PASTE HERE
```
</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020):
+ Installed LaTeX packages:

</details>

<details><summary>FFMPEG</summary>

Output of `ffmpeg -version`:

```
PASTE HERE
```
</details>

### PR DESCRIPTION
**Pull request by sparshg**
## Overview:

Fixed Rotation animation when trying to rotate by multiples of 2π
Closes #1935 

## Explanation

Multiples of 2π didn't work because the start and end position \*almost\* coincides, so path function wasn't completely accurate, and interpolation was wobbly, ~~wrote custom `path_func` for rotation animation to fix~~

https://user-images.githubusercontent.com/43041139/130240719-57127feb-1792-4f24-8996-d9f2d343d206.mp4

https://user-images.githubusercontent.com/43041139/130240662-d8512e97-8e6f-417e-a7f9-fe629552efd6.mp4

 

## Reviewer Checklist
- [x] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [x] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by sparshg** [comment]
well that was an annoying bug ever since I started using manim lol, never noticed it only happens at multiples of 2π

**Comment by behackl** [comment]
See my comment about this in #1935, I think it would be better to fix the underlying issue with `path_arc`.

**Comment by sparshg** [comment]
I saw that, but that cannot be fixed because that function takes the start and end point only and generates an arc, which is not possible if both points coincide, (because radius cannot be known). Here tho we know the radius so thats why I think writing custom function is better.

**Comment by sparshg** [comment]
hmmmm ook lemme try

**Comment by sparshg** [comment]
ok so I tried this, seems better imo

**Comment by sparshg** [comment]
ok so how about this

File: manim/animation/transform.py [hunk 1]
```diff
@@ -71,7 +73,9 @@ def path_arc(self) -> float:
     @path_arc.setter
     def path_arc(self, path_arc: float) -> None:
         self._path_arc = path_arc
-        self._path_func = path_along_arc(self._path_arc, self.path_arc_axis)
+        self._path_func = path_along_arc(
+            self._path_arc, self.path_arc_axis, self.path_arc_centers
```
Comment (behackl) [inline manim/animation/transform.py:77]: I have a (slight) preference for passing keyword arguments as actual keyword arguments.

```suggestion
            self._path_arc,
            axis=self.path_arc_axis,
            arc_centers=self.path_arc_centers
```

**Review by behackl — CHANGES_REQUESTED** [review]
Looks good to me. Could you add an animated graphical unit test (see `tests/test_graphical_units/test_transform.py` for examples how this should look) for a scene that has four squares that rotate (in the same play call) by -4pi, -2pi, 2pi, 4pi?

Apart from the missing test, this is good to go from my point of view.

**Review by behackl — APPROVED** [review]
Looks good to me! Thanks for fixing this.

**Pull request merged**